helix-common/src/main/java/org/apache/helix/SystemPropertyKeys.java

```diff
@@ -82,4 +82,6 @@
   // System Property Metadata Store Directory Server endpoint key
   public static final String MSDS_SERVER_ENDPOINT_KEY =
       MetadataStoreRoutingConstants.MSDS_SERVER_ENDPOINT_KEY;
+
+  public static final String STATEUPDATEUTIL_ERROR_LOG2ZK_ENABLED = "helix.StateUpdateUtil.errorLog.enabled";
```

Comment:
I thought more about this configuration. Can we directly make it an enum now instead of changing it later? Otherwise, we will need to deprecate this item and go through the painful backward compatibility checklist in the future.

I suggest making it STATEUPDATE_ERROR_REPORTING_CHANNEL.

Reply:
Can you illustrate the idea how to make it enum? Something like 

```
enum STATEUPDATE_ERROR_REPORTING_CHANNEL {
    LOG4J,
    LOG2ZK
};

```

and in the system property, 
`public static final String STATEUPDATEUTIL_ERROR_LOG2ZK_ENABLED = <enum value>`;
then we will convert the system property to enum value?

Reply:
STATEUPDATEUTIL_ERROR_PERSISTANCY_ENABLED

Reply:
changed.